### PR TITLE
[AIRFLOW-5643] Reduce duplicated logic in S3Hook

### PR DESCRIPTION
**Pull request by louisguitton**
### Jira

- [x] https://issues.apache.org/jira/browse/AIRFLOW-5643

### Description

- [x] `load_bytes` is a convenience method that delegates to `load_fileobj`, just like `load_string` delegates to `load_bytes`. This change is here to make this logic more apparent.

### Tests

- [x] No new functionality introduced

```
./breeze --backend postgres --test-target tests.hooks.test_s3_hook
```

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] No new functionality introduced

### CODE REVIEW

**Review by feluelle — COMMENTED** [review]
_Just my opinion:_ I rather create a private function that will be called in public functions. So that you do not rely on another public function (in case the public function gets changed).

**Review by milton0825 — APPROVED** [review]

**Comment by louisguitton** [comment]
Thank you both for your reviews. I agree with you @feluelle , I'll make the suggested changes

**Review by feluelle — APPROVED** [review]
Nice!

**Comment by feluelle** [comment]
I restarted the failed travis checks.

**Comment by potiuk** [comment]
If those are the kubernetes ones - maybe merge it without waiting. We have found with @gerardo a reason why kubernetes tests might timeout and we will fix it shortly - but I do not want to hold it back!

File: airflow/hooks/S3_hook.py [hunk 1]
```diff
@@ -487,20 +485,8 @@ def load_bytes(self,
...
-            raise ValueError("The key {key} already exists.".format(key=key))
-
-        extra_args = {}
-        if encrypt:
-            extra_args['ServerSideEncryption'] = "AES256"
-
-        filelike_buffer = io.BytesIO(bytes_data)
-
-        client = self.get_conn()
-        client.upload_fileobj(filelike_buffer, bucket_name, key, ExtraArgs=extra_args)
```
Comment (ashb) [inline airflow/hooks/S3_hook.py:503]: This appears to have removed extra_args!

**Review by ashb — CHANGES_REQUESTED** [review]

File: airflow/hooks/S3_hook.py (hunk 1 above)
Comment (ashb) [reply airflow/hooks/S3_hook.py:503]: Oh no I see, I thought at first this was coming from user input but it doesn't.

**Review by ashb — APPROVED** [review]

**Comment by louisguitton** [comment]
> If those are the kubernetes ones - maybe merge it without waiting. We have found with @gerardo a reason why kubernetes tests might timeout and we will fix it shortly - but I do not want to hold it back!

Yeah the 2 failing tests are the k8s ones :/

**Comment by feluelle** [comment]
Yes, Jarek I just saw the thread right after I reran the tests. 

I think we can merge it then. 👍

**Pull request merged**